dashboard/public/pixelation/pixelation.js

```diff
@@ -385,3 +385,25 @@ function showPNG() {
   w.focus();
   options.saveProject && options.saveProject();
 }
+
+function onFinishedButtonClick() {
+  var finishedButton = $('#finished');
+  if (finishedButton.attr('disabled')) {
+    return;
+  }
+  finishedButton.attr('disabled', true);
+
+  if (options.saveProject) {
+    options.saveProject(onSaveProjectComplete);
```

Comment:
I wonder if in the future it would make sense to have dashboard.project.save return a promise, allowing us to do something like this instead (which feels like it shows what's happening a little bit more clearly to me).
```
options.saveProject().then(onSaveProjectComplete).
```
We could also promisify process results, and get something like
```
options.saveProject()
  .then(function () {
    return processResults();
  })
  .then(function (response) {
    if (!response.nextRedirect) {
      finishedButton.attr('disabled', false);
    }
  });
```

If we want to dream even bigger, we could use ES6 arrow functions
```
options.saveProject()
  .then(() =>  processResults())
  .then(response =>  {
    if (!response.nextRedirect) {
      finishedButton.attr('disabled', false);
    }
  });